fiftyone/core/dataset.py

```diff
@@ -4350,12 +4370,26 @@ def _clear_frames(self, view=None, sample_ids=None, frame_ids=None):
 
             sample_ids = view.values("id")
 
+        ops = []
         if sample_ids is not None:
-            d = {"_sample_id": {"$in": [ObjectId(_id) for _id in sample_ids]}}
+            batch_size = fou.recommend_batch_size_for_value(
```

Comment:
Just a thought, what if we can do a refactor, something like `dataset._delete(...)` where we can decide which code path to use based on the number of documents? Is there any advantage to using `self._sample_collection.delete_many`?

Reply:
I admittedly haven't benchmarked thoroughly, but I'd expect that `delete_many(...)` and `bulk_write([DeleteMany(...)], ...)` are basically equivalent.